packages/core/src/components/Dropdown/Dropdown.jsx

```diff
@@ -78,14 +89,9 @@ export class Dropdown extends React.PureComponent {
           className={fieldClasses}
           id={this.id()}
           /* eslint-disable no-return-assign */
-          ref={ref => {
-            if (focusTrigger) {
-              this.selectRef = ref;
-            }
-            if (fieldRef) {
-              fieldRef(ref);
-            }
-          }}
+          ref={
+            focusTrigger ? ref => (this.focusRef = ref) : inputRef || fieldRef
```

Comment:
This line is more confusing to read than what was there before. There's a ternary, an anonymous function with an assignment statement for its return value as well as an OR operator. My eyes can't easily resolve what's happening.

Reply:
Good point, hows this instead?
```
          ref={ref => {
            if (focusTrigger) {
              this.focusRef = ref;
            } else {
              if (inputRef) {
                inputRef(ref);
              }
              if (fieldRef) {
                fieldRef(ref);
              }
            }
          }}
```

Reply:
Yep, that looks good to me 👍 It's certainly longer, but the logic is clearer to me.